### ISSUE TITLE
How can I debug server typescript code in VS Code?

### ISSUE THREAD

**Issue by aleho70** [mean-expert-official/fireloop.io#67]

**Comment by jonathan-casarrubias** [comment]
Hi @aleho70 thanks for reaching out... hey so that is a good question, I personally have not tried to debug the server using VCode, though is good to start reviewing some ways to do it.

For now I apologize not to give a real answer on this, will need to verify.

Regards,
Jon

**Comment by hellraiserii** [comment]
Hey! I was in the same situation and while it was not an obvious solution, it worked like a charm.
TL;DR: add `"protocol": "inspector"` in a vanilla node.js launch program configuration.

Elaborating the answer a bit more, and as far as I could understand, fireloop registers ts-node in order to compile in memory all your .ts files on demand, leaving no resulting .js files nor source-mapping files on disk. This was a problem for VSCode originally, but it seems they managed to support in memory source maps by means of a new "inspector" protocol. I googled a lot without success. 
But finally, the solution appeared almost by chance... I dumbly searched for **"tag:debuggers node"** inside VSCode and TWO builtin debuggers appeared. I followed the instructions to use the second one that implements Chrome Debugging Protocol. This plugin is now outdated but they merged it with the main one, and its by setting "protocol":"inspector" that this extension gets enabled.
So, if you create a vanilla node launch configuration, and add this property, you should be able to set breakpoints on your .ts files and be as happy as I am with it :D.
{
    "version": "0.2.0",
    "configurations": [{
        "type": "node",
        "request": "launch",
        **"protocol": "inspector",**
        "name": "Launch Program",
        "program": "${workspaceRoot}/server/server.js",
        "outFiles": [
            "${workspaceRoot}/out/**/*.js"
        ]
    }]
}
I'm sorry if my answer is misleading at any point. I hope this solution works for you too.
